backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/UsersController.java

```diff
@@ -25,26 +35,75 @@
+	private final SubscriptionModelMapper mapper;
+
+	private final AlertTypeService alertTypeService;
+
+	public UsersController(
+				SubscriptionModelAssembler subscriptionModelAssembler, 
+				SubscriptionService subscriptionService, 
+				SubscriptionModelMapper mapper, 
+				AlertTypeService alertTypeService) {
 		Assert.notNull(subscriptionModelAssembler, "subscriptionModelAssembler is required; it must not be null");
 		Assert.notNull(subscriptionService, "subscriptionService is required; it must not be null");
+		Assert.notNull(mapper, "mapper is required; it must not be null");
+		Assert.notNull(alertTypeService, "alertTypeService is required; it must not be null");
 		this.subscriptionModelAssembler = subscriptionModelAssembler;
 		this.subscriptionService = subscriptionService;
+		this.mapper = mapper;
+		this.alertTypeService = alertTypeService;
 	}
 
-	@GetMapping({ "/{id}/subscriptions" })
+	@GetMapping({ "/{userId}/subscriptions" })
 	@Operation(summary = "Get all subscriptions for a user.", operationId = "get-user-subscriptions")
 	public CollectionModel<SubscriptionModel> getSubscriptionsByUserId(
-			@NotBlank(message = "id must not be null or blank")
-			@Parameter(description = "The id of the user.", example = "000000000", required = true)
-			@PathVariable String id) {
-		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(id));
-		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(id)).withSelfRel();
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId) {
+		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(userId));
+		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(userId)).withSelfRel();
 		return subscriptionModelAssembler.wrapCollection(subscriptions, SubscriptionModel.class).add(selfLink);
+	}	
+
+
+	@PostMapping({ "/{userId}/subscriptions" })
+	@Operation(summary = "Create a subscription for a user.", operationId = "user-subscriptions-create")
+	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
+	public void create(
+			@RequestBody(required = true)
+			SubscriptionModel createSubscriptionRequest,
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId) throws Exception {
+		final Optional<AlertType> convertedAlertType = Optional.ofNullable(alertTypeService.readByCode(createSubscriptionRequest.getAlertType())).orElseThrow(() -> new Exception("alertType not found"));
+		createSubscriptionRequest.setAlertType(convertedAlertType.get().getId());
```

Comment:
I don't think you should mutate the incoming request. Instead, update the mapper to allow you to send the alert id in.. like this:

``` java
final var subscription = mapper.toDomain(subscription, alertTypeId);
```

Reply:
Ideally all of the REST models should be immutable, but I haven't quite figured out how to do that yet (because of spring hateoas).

Reply:
fixed